### PR TITLE
[subset] Add HB_SUBSET_FLAGS_IFTB_REQUIREMENTS flag.

### PR DESCRIPTION
**Pull request by garretrieger**
When enabled the output subset conforms to the requirements for a base font to be patched by IFTB patches. IFTB is a proposed incremental font transfer patch. This new flag is set as experimental. Currently, setting this flag causes the output subset to also use long offsets for outline data in loca/glyf, gvar, CFF, and CFF2.

This change is a version of https://github.com/skef/harfbuzz/commit/3ae2fe2084d96f61d8ad1bbe42a6ef4e62ec830a rebased onto head w/ tests added.

### CODE REVIEW

File: src/hb-ot-var-gvar-table.hh [hunk 1]
```diff
@@ -428,7 +428,10 @@ struct gvar
       subset_data_size += get_glyph_var_data_bytes (c->source_blob, glyph_count, old_gid).length;
     }
 
-    bool long_offset = subset_data_size & ~0xFFFFu;
+    bool long_offset = (subset_data_size & ~0xFFFFu);
+ #ifdef HB_EXPERIMENTAL_API
+    long_offset = long_offset || (c->plan->flags & HB_SUBSET_FLAGS_IFTB_REQUIREMENTS);
```
Comment (behdad) [inline src/hb-ot-var-gvar-table.hh:433]: I think `||=` is a valid op?

File: src/hb-ot-cff-common.hh [hunk 2]
```diff
@@ -111,11 +112,14 @@ struct CFFIndex
...
   bool serialize_header (hb_serialize_context_t *c,
 			 Iterator it,
-			 unsigned data_size)
+			 unsigned data_size,
+                         unsigned min_off_size = 0)
   {
     TRACE_SERIALIZE (this);
 
     unsigned off_size = (hb_bit_storage (data_size + 1) + 7) / 8;
+    if (min_off_size > off_size)
```
Comment (behdad) [inline src/hb-ot-cff-common.hh:121]: Maybe `hb_max`?

File: src/hb-subset-cff1.cc [hunk 3]
```diff
@@ -778,13 +786,42 @@ struct cff1_subset_plan
...
+
+  plan.info.char_strings_link = c->pop_pack (false);
+  return true;
+}
+
 bool
 OT::cff1::accelerator_subset_t::serialize (hb_serialize_context_t *c,
 					   struct OT::cff1_subset_plan &plan) const
 {
+  if (plan.charstrings_last) {
```
Comment (behdad) [inline src/hb-subset-cff1.cc:820]: I don't see any issue always putting the charstrings last. (with a comment obviously).

**Comment by behdad** [comment]
Left a couple of comments. LGTM otherwise.

File: src/hb-ot-cff-common.hh (hunk 2 above)
Comment (garretrieger) [reply src/hb-ot-cff-common.hh]: Done

File: src/hb-subset-cff1.cc (hunk 3 above)
Comment (garretrieger) [reply src/hb-subset-cff1.cc:820]: Sounds good, I'll do that.

File: src/hb-ot-var-gvar-table.hh (hunk 1 above)
Comment (garretrieger) [reply src/hb-ot-var-gvar-table.hh:433]: Doesn't look like it. I think only |= is defined.

**Pull request merged**